src/main/java/org/folio/circulation/services/feefine/FeeTypeBasedRefundProcessor.java

```diff
@@ -0,0 +1,135 @@
+package org.folio.circulation.services.feefine;
+
+import static java.util.Arrays.asList;
+import static org.folio.circulation.domain.AccountRefundReason.LOST_ITEM_FOUND;
+import static org.folio.circulation.domain.FeeAmount.zeroFeeAmount;
+import static org.folio.circulation.domain.FeeFine.LOST_ITEM_FEE_TYPE;
+import static org.folio.circulation.domain.FeeFine.LOST_ITEM_PROCESSING_FEE_TYPE;
+import static org.folio.circulation.domain.representations.AccountPaymentStatus.CANCELLED_ITEM_RETURNED;
+import static org.folio.circulation.domain.representations.AccountPaymentStatus.CREDITED_FULLY;
+import static org.folio.circulation.domain.representations.AccountPaymentStatus.REFUNDED_FULLY;
+import static org.folio.circulation.domain.representations.StoredFeeFineAction.StoredFeeFineActionBuilder;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.folio.circulation.domain.Account;
+import org.folio.circulation.domain.AccountRefundReason;
+import org.folio.circulation.domain.FeeAmount;
+import org.folio.circulation.domain.representations.AccountPaymentStatus;
+import org.folio.circulation.domain.representations.StoredFeeFineAction;
+
+public class FeeTypeBasedRefundProcessor implements AccountRefundProcessor {
```

Comment:
What does `fee type based` refer to? Is it that refunds are based upon the type? 

If so, is that the types of actions involved in the handling of the fee?

Reply:
It means that refund is based on `FeeFine type`. The fee/fine type defines `refund reason` of an account and `close` status.

For example for lost item fees:
* refund reason is - lost item found;
* close status u is - Cancelled lost item found;

Reply:
Ok, thank you.

Are there other ways to determine how to refund the paid part of a fee?

If not, I suggest simplifying the name of this to `FeeRefundProcessor` (I might also like to remove the processor term, however I don't know how yet).

Reply:
> Are there other ways to determine how to refund the paid part of a fee?

I think no, it should be generalized for all fee types.

Reply:
The class renamed to `FeeRefundProcessor`